### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by Vova-Stoliar**
[DEMO LINK](https://Vova-Stoliar.github.io/react_dynamic-list-of-goods/)

### CODE REVIEW

File: src/api/goods.ts [hunk 1]
```diff
@@ -1,11 +1,26 @@
...
-export const getRedGoods = () => {};
+  return response.json();
+};
+
+export const get5First = () => {
+  const showFiveGoods = getAll().then(goods => goods
+    .sort((a, b) => a.name.localeCompare(b.name))
+    .slice(0, 5));
+
+  return showFiveGoods;
```
Comment (Elena-brv) [inline src/api/goods.ts:19]:
```suggestion
 return getAll()
   .then(goods => goods
     .sort((a, b) => a.name.localeCompare(b.name))
     .slice(0, 5)
   );
```

File: src/App.tsx [hunk 2]
```diff
@@ -1,12 +1,75 @@
...
-);
+export class App extends React.PureComponent<{}, State> {
+  state:State = {
+    goods: [{
+      id: NaN,
+      name: '',
+      color: '',
+    }],
+    showGoods: false,
+    showError: false,
```
Comment (Elena-brv) [inline src/App.tsx:20]: is it not enough only one of these properties?

File: src/App.tsx [hunk 3]
```diff
@@ -1,12 +1,75 @@
...
+      <>
+        <h1>Dynamic list of Goods</h1>
+        <div>
+          <button
+            type="button"
+            onClick={() => this.handleButton(getAll())}
+          >
+            Load All goods
+          </button>
+          {' '}
```
Comment (Elena-brv) [inline src/App.tsx:51]: why do you need this?

File: src/App.tsx [hunk 4]
```diff
@@ -1,12 +1,75 @@
...
+          <button
+            type="button"
+            onClick={() => this.handleButton(getRedGoods())}
+          >
+            Load red goods
+          </button>
+        </div>
+        {this.state.showGoods && <GoodsList goods={this.state.goods} />}
+        {this.state.showError && (
```
Comment (Elena-brv) [inline src/App.tsx:67]: you can use only one of these properties and use ternary operator

**Review by Elena-brv — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 3 above)
Comment (Vova-Stoliar) [reply src/App.tsx:47]: to have tiny space between buttons

Comment (Elena-brv) [reply src/App.tsx:47]: use css for this

**Review by Elena-brv — COMMENTED** [review]
image

fix the API link

**Pull request closed without merging**